### PR TITLE
config/config: add instantiation helper methods from jet

### LINKED ISSUE #18861: Add configuration instantiation helper methods from jet
Jet had these - https://jet-start.sh/docs/operations/configuration#configuration-loaders
They were removed in the merge with IMDG. All is achievable using other existing public API but it's more verbose. For the sake of completeness and breaking less jet user code (as reported on community slack - https://hazelcast.slack.com/archives/C01JU7ZJYGP/p1623160436317200?thread_ts=1623159067.316600&cid=C01JU7ZJYGP), we may add these helper methods.

### PR DESCRIPTION
**Pull request by fbarotov**
Restore non-deprecated static config constructors from Jet 

Fixes #18861

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/Config.java [hunk 1]
```diff
@@ -207,32 +214,254 @@ public Config(String instanceName) {
...
+
+    /**
+     * Loads Config using the default {@link #load() lookup mechanism} to locate the configuration
+     * file, except it does not override configuration properties using the environment variables
+     * or system properties.
+     *
+     * @return Config created from a file when exists, otherwise default.
+     */
+    public static Config loadDefault() {
+        return load(System.getProperties());
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/config/Config.java:257]: Nicely done! 🏅 

Can you just make sure that all variants of the helper methods use `ExternalConfigurationOverride().overwriteMemberConfig()`, as it's a really useful feature that allows user to override most config without modifying XML or YAML files (https://github.com/hazelcast/hazelcast/pull/17348)?

Comment (fbarotov) [reply hazelcast/src/main/java/com/hazelcast/config/Config.java]: thanks :) done

**Review by jbartok — APPROVED** [review]

**Review by blazember — COMMENTED** [review]
I think we should add tests that verify that the external config overrides take place. See `ExternalMemberConfigurationOverrideEnvTest` as an example.

FYI, we can even simplify further the utility methods by removing the XML/YAML indicators from the method names. We have the `ConfigRecognizer` API with existing implementations we can build upon. The existing implementations sample the head of the string/stream to decide if the provided config is "recognized" as a member/client/client-failover declarative config described in XML or YAML languages, so based only on the top-level tags. Now we have schema validation for both languages and we can be even more accurate. Not saying we should definitely do this, but if we agree so, 5.0 is a good excuse to do that. cc @mmedenjak

**Comment by mmedenjak** [comment]
Thinking... 🤔 

Yes, let's do it. It'll make the API simpler I guess and possibly moving from the ex-jet API to the new unified methods for config instantiation won't be too hard. You just replace the method names and (hopefully), our implementation figures it out and parses it the right way.

So: `loadXmlFromString`,`loadYamlFromString` -> `loadFromString`.

@jbartok WDYT? Any objections from the "jet team"? :)

**Comment by mmedenjak** [comment]
@blazember can you re-review?

File: hazelcast/src/main/java/com/hazelcast/config/Config.java [hunk 2]
```diff
@@ -207,32 +216,228 @@ public Config(String instanceName) {
...
+     *
+     * @param source the XML or YAML stream
+     * @param properties properties to use for variable resolution
+     * @throws com.hazelcast.core.HazelcastException if the source content is invalid
+     * @return Config created from the stream
+     */
+    public static Config loadFromStream(InputStream source, Properties properties) {
+        Config cfg = null;
+        try {
+            cfg = loadFromConfigStream(new ConfigStream(source), properties);
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/config/Config.java:418]: I believe we might invalidate the config here if it is more than 4K. The `ConfigStream` was designed to protect against attacks with untrusted streams, therefore we have this limitation. So I would pass the original stream down and if the `ConfigStream` wrapping it is recognized, we can build the config from the original stream, not from `ConfigStream`. Not sure if we have the same concerns with the original stream where this `Config.load()` can be used, probably not.

Alternatively, we can make the used size of the `ConfigStream` configurable via sysprops, but I think that's not really a good option.

Comment (fbarotov) [reply hazelcast/src/main/java/com/hazelcast/config/Config.java]: Thanks a lot :) It is fixed now.

**Review by blazember — APPROVED** [review]
LGTM :clap:

**Pull request merged**